src/estimagic/optimization/history.py

```diff
@@ -0,0 +1,247 @@
+"""History class for pounders and similar optimizers."""
+import numpy as np
+
+
+class LeastSquaresHistory:
+    """Container to save and retrieve history entries for a least-squares optimizer."""
```

Comment:
Looking at this class alone its not entirely clear what a "history" is. Maybe add one or two sentences to the docstring.